onnxruntime/core/session/onnxruntime_c_api.cc

```diff
@@ -615,6 +638,37 @@ ORT_API_STATUS_IMPL(OrtApis::RegisterCustomOpsLibrary, _Inout_ OrtSessionOptions
   API_IMPL_END
 }
 
+ORT_API_STATUS_IMPL(OrtApis::RegisterCustomOpsLibrary_V2, _Inout_ OrtSessionOptions* options, _In_ const char* library_path) {
+  API_IMPL_BEGIN
+  void* library_handle = nullptr;
+
+  ORT_API_RETURN_IF_STATUS_NOT_OK(Env::Default().LoadDynamicLibrary(library_path, false, &library_handle));
+  if (!library_handle)
+    return OrtApis::CreateStatus(ORT_FAIL, "RegisterCustomOpsLibrary_V2: Failed to load library");
+
+  std::shared_ptr<void> lib_ptr(library_handle,
+      [](void* handle) {
+        auto status = Env::Default().UnloadDynamicLibrary(handle);
+        if (!status.IsOK()) {
+          LOGS_DEFAULT(WARNING) << "RegisterCustomOpsLibrary_V2: Failed to unload custom op library.";
+        }
+      });
```

Comment:
Let's consider this custom destruction in the context of the SessionOptions destruction only and not anywhere else.